cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/persisteddata/namespace/ExecutorPersistedData.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2024 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+package com.linkedin.kafka.cruisecontrol.persisteddata.namespace;
+
+import com.linkedin.kafka.cruisecontrol.executor.Executor;
+import com.linkedin.kafka.cruisecontrol.persisteddata.ValueMappedMap;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+import org.apache.commons.lang3.tuple.Pair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Persisted data used by an {@link Executor} instance.
+ */
+public class ExecutorPersistedData {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ExecutorPersistedData.class);
+
+    /**
+     * Keys to reference the particular broker map. For use with the "get" and "set" methods.
+     */
+    public enum BrokerMapKey {
+        /**
+         * Key to reference the "demote broker" map.
+         */
+        DEMOTE,
+
+        /**
+         * Key to reference the "remove broker" map.
+         */
+        REMOVE;
+    }
+
+    // Holds the maps for the demoted brokers and the removed brokers.
+    private final Map<String, Map<Integer, Long>> _demotedOrRemovedBrokers;
+
+    // Used to make thread safe changes to _demotedOrRemovedBrokers.
+    private final Map<BrokerMapKey, Object> _locks = Map.of(
+            BrokerMapKey.DEMOTE, new Object(),
+            BrokerMapKey.REMOVE, new Object()
+    );
+
+    /**
+     * Executor data that needs persisting is stored in the given map. It assumes that the map's
+     * data is persisted independently. This class is concerned with providing a clean view of that
+     * data and not with how it is stored.
+     *
+     * @param persistedMap The map to store {@link Executor} data in.
+     */
+    public ExecutorPersistedData(Map<String, String> persistedMap) {
+        this._demotedOrRemovedBrokers = Namespace.EXECUTOR.embed(
```

Comment:
If I understand correctly, you do this because this way we can serialize demoted or removed brokers with their timestamp the same way in both cases (in-memory vs Kafka).
To be completely honest, I find this a bit less intuitive to understand at first. For this I'd like to propose that what if we move the type of serialization to the respective storage method? For instance we could have a class like this:
```
BrokerEvent implements Serializable {
  private String brokerId;
  private long timestamp;
  private EventType eventType; // demote or remove
  private Namespace namespace; // currently "executor" only
}
```
For memory serialization we could use the standard ByteArrayInput/OutputStream serialization while for Kafka topics we could use json that is serialized into Strings. That would make the Kafka topics easily debuggable too.
When deserialized, we can represent them in a Map<String, BrokerEvent> structure to map broker ID to the event to make it easier to query.
Let me know your thoughts.